asset-tracking-history-saver/src/index.ts

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env node
+import * as Ably from 'ably/promises';
+import { Command } from 'commander';
+import { CHANNEL_PREFIX, CLIENT_ID } from './consts';
+import { getEnhancedLocationsFromChannelHistory } from './getDataFromChannelHistory';
+import { getDataFromFile } from './getDataFromFile';
+import { saveDataToFile } from './saveDataToFile';
+import { AblyCredentialsFileData, ConfigurationFileData, EnhancedLocationUpdate } from './types';
+const onProcessInterrupted = require('death');
+const VERSION = require('../package.json').version;
+
+const program = new Command();
+
+program.version(VERSION);
+
+program.requiredOption('-c, --configuration <file>', 'configuration file');
+program.option('-v, --verbose', 'enable verbose logging');
+
+program.parse(process.argv);
+
+const opts = program.opts();
+
+const configurationData: ConfigurationFileData = getDataFromFile(opts.configuration);
+if (
+  !configurationData.sourceCredentials ||
+  !configurationData.sourceTrackableId ||
+  !configurationData.destinationFile
+) {
+  console.error('You need to specify source credentials, source trackable ID and destination file path.');
+  process.exit(1);
```

Comment:
Not really a dealbreaker but I would generally advise to avoid using `process.exit(1)` in favour of just throwing an exception - this makes it possible to catch and handle errors for stuff like testing.

Reply:
I replaced the `process.exit(1)` with throwing an error https://github.com/ably/ably-asset-tracking-common/pull/43/commits/52291b9a47273e43ee4b7bd45d07afdf2d8f5cb5 (only in the interruption handler it wasn't possible as an exception there didn't stop the process)

Additionally, I removed the `process.exit(0)` from the end of the script, as it seems to do nothing :wink: https://github.com/ably/ably-asset-tracking-common/pull/43/commits/da44adb922ac613ca32bb21437fef3d3ea2d4584